src/NSwag.AspNet.Owin/Middlewares/OpenApiDocumentMiddleware.cs

```diff
@@ -44,7 +44,7 @@ public OpenApiDocumentMiddleware(OwinMiddleware next, string path, IEnumerable<T
         /// <returns>The task.</returns>
         public override async Task Invoke(IOwinContext context)
         {
-            if (context.Request.Path.HasValue && string.Equals(context.Request.Path.Value.Trim('/'), _path.Trim('/'), StringComparison.OrdinalIgnoreCase))
+            if (context.Request.Path.HasValue && string.Equals(context.Request.Path.Value, _path, StringComparison.OrdinalIgnoreCase))
```

Comment:
What if `context.Request.Path.Value` doesn't start with `'/'` here?

Reply:
Is that possible?

Either way, creating new strings is not the solution.

Reply:
I think it is possible in general (AFAIR it can be empty string for example). I do agree with you, low allocation code is good, I'm just saying that this seem to be changing comparison logic.

Reply:
I think this looks fine (did you test it locally)?
I'd say we merge this... ok?

Reply:
I only tested on ASP.NET Core.

But the path is supposed to always start with `7`, as far as I recall.

Reply:
Ok then..